nucypher/network/protocols.py

```diff
@@ -39,6 +39,14 @@ def parse_node_uri(uri: str):
     else:
         checksum_address = None  # federated
 
+    #############################################
+    # Strange logic here to ensure https:// - possibly pursuant to https://bugs.python.org/msg179670
```

Comment:
Strange indeed - this was just a circle ci issue, no? If so, then perhaps you can temporarily wrap this strange logic with a circle ci check

```python
if 'CIRCLECI' in os.environ:
   ...
```

Reply:
Well, presumably it will happen anywhere that the affected python versions are used, so it's probably best to just keep it environ agnostic.  Hopefully we disappear it soon anyway.